### ISSUE TITLE
Cursor key detected as Escape on iTerm2

### ISSUE THREAD

**Issue by max-arnold** [Textualize/textual#5251]
The issue happens on iTerm2 under macOS, and doesn't happen in the stock Terminal app. When I press and hold the down arrow key, the minimal Textual application (see below) exits and prints `^[[B` in the terminal. Looks similar to https://github.com/Textualize/textual/issues/4816.

![2024-11-18 11 12 17](https://github.com/user-attachments/assets/56a869b0-1246-45ce-b866-0cc5fbc8536d)

```python
from textual.app import App

class MyApp(App):
    BINDINGS = [
        ("escape", "quit()", "Quit"),
    ]

if __name__ == "__main__":
    app = MyApp()
    app.run()
```

Tried resetting iTerm2 settings to default values by deleting the `~/Library/Preferences/com.googlecode.iterm2.plist`, but the issue still persists.

```
# Textual Diagnostics

## Versions

| Name    | Value  |
|---------|--------|
| Textual | 0.86.1 |
| Rich    | 13.9.4 |

## Python

| Name           | Value                               |
|----------------|-------------------------------------|
| Version        | 3.12.6                              |
| Implementation | CPython                             |
| Compiler       | Clang 14.0.3 (clang-1403.0.22.14.1) |
| Executable     | /private/tmp/test/.venv/bin/python3 |

## Operating System

| Name    | Value                                                                                                  |
|---------|--------------------------------------------------------------------------------------------------------|
| System  | Darwin                                                                                                 |
| Release | 22.6.0                                                                                                 |
| Version | Darwin Kernel Version 22.6.0: Thu Sep  5 20:48:48 PDT 2024; root:xnu-8796.141.3.708.1~1/RELEASE_X86_64 |

## Terminal

| Name                 | Value             |
|----------------------|-------------------|
| Terminal Application | iTerm.app (3.5.9) |
| TERM                 | xterm-256color    |
| COLORTERM            | truecolor         |
| FORCE_COLOR          | *Not set*         |
| NO_COLOR             | *Not set*         |

## Rich Console options

| Name           | Value               |
|----------------|---------------------|
| size           | width=80, height=25 |
| legacy_windows | False               |
| min_width      | 1                   |
| max_width      | 80                  |
| is_terminal    | True                |
| encoding       | utf-8               |
| max_height     | 25                  |
| justify        | None                |
| overflow       | None                |
| no_wrap        | False               |
| highlight      | None                |
| markup         | None                |
| height         | None                |
```

**Comment by max-arnold** [comment]
The issue doesn't happen in Textual 0.75.1 and starts to manifest itself in 0.76.0

**Comment by darrenburns** [comment]
I think this is a bug in recent versions of iTerm. Lots of people (myself included), have been having similar issues.

**Comment by willmcgugan** [comment]
Installing the nightly here seemed to fix it for me https://iterm2.com/downloads.html

**Comment by max-arnold** [comment]
Thanks! Found an existing issue https://gitlab.com/gnachman/iterm2/-/issues/11982

iTerm2-3.5.10 fixes it.

<img width="621" alt="CleanShot 2024-11-19 at 07 29 16@2x" src="https://github.com/user-attachments/assets/57f8baa6-51d1-4e95-a215-d30f0d800b10">

**Comment by finger563** [comment]
FWIW i'm still getting this issue on the latest iterm nightly after exiting my textualize program

**Comment by darrenburns** [comment]
@finger563 what version of Textual are you using? You may need to update.

**Comment by finger563** [comment]
@darrenburns I'm using 0.81.0, as anything later and I have a similar issue with windows terminals when using my CI to package my app into an executable with pyinstaller (https://github.com/Textualize/textual/issues/5162)

**Comment by darrenburns** [comment]
Could you confirm on the latest version of Textual that you still get both of these issues? There have been changes very recently in areas that may be related.

**Comment by finger563** [comment]
@darrenburns Just upgraded to latest iTerm2 and Textual (3.4.0) and I'm still getting this issue.

**Comment by finger563** [comment]
@darrenburns I was able to fix it (though i'm not actually sure of the other consequences 😅 ) by setting iTerm2 to `report keys using CSI u`. I had to restart my terminals, but after doing that I'm able to use a textualize program, exit it, and have my ctrl+ keybindings all work.

![Image](https://github.com/user-attachments/assets/ade4ab1f-2561-47ab-a7f1-a164bd8d21ce)

Based on this, it does seem that it's possibly something `textual` can fix when it exits, by disabling CSI u, which it seems to be enabling by default when it runs.

Edit: 
https://gitlab.com/gnachman/iterm2/-/issues/10869 this issue helped me find this potential fix / information.
